pgjdbc/src/main/java/org/postgresql/core/NativeQuery.java

```diff
@@ -27,13 +28,14 @@
   }
 
   public NativeQuery(String nativeSql) {
-    this(nativeSql, NO_BINDS);
+    this(nativeSql, NO_BINDS, false);
   }
 
-  public NativeQuery(String nativeSql, int[] bindPositions) {
+  public NativeQuery(String nativeSql, int[] bindPositions, boolean compatible) {
```

Comment:
See issue #488 

I've recommended we pull out all of the logic related to SQL parsing and turn it into its own PR.  The main concern here is that we've gone to the trouble of identifying keywords in the SQL statement but Parser.java discards everything its learned after implementing "business logic" as to the limited scope of whether a given query is "rewrite compatible" (minor nitpick - "compatible" is not specific enough a label).  The execution layer should be the one to make these kinds of determinations.  Since we don't have or need a true re-write layer - the executor can attempt the rewrite before sending the query if it determines that the nature of the query will support doing so.

A better approach would to set to attributes/properties as to statement major type (SELECT, INSERT, DELETE, UPDATE, DDL) and secondary attributes (e.g. RETURNING - yes/no/n-a) during parse and define a public API for inspecting those properties during execution or wherever else logic depends on the query - like in 488.

David J.